homeassistant/components/mqtt/light/schema_json.py

```diff
@@ -406,6 +406,9 @@ def state_received(msg: ReceiveMessage) -> None:
                         values["color_temp"],
                         self.entity_id,
                     )
+                # Allow to switch back to color_temp
+                if "color" not in values:
+                    self._attr_hs_color = None
```

Comment:
With legacy color handling `_attr_hs_color` is used to determine the color mode, so we should reset it if we switch to `color_temp`:

https://github.com/home-assistant/core/pull/105283/files#diff-cf35ae315d8066bc8f6fe286f36504d3f522cf3d6c7088af1ab409756b2badc2R459-L467